src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -227,3 +221,62 @@ object FilterIndexRule extends Rule[LogicalPlan] with Logging {
     }
   }
 }
+
+object FilterRuleExtractor extends Logging {
+  type returnType = (
+      LogicalPlan,
+      Filter,
+      Seq[String], // output columns
+      Seq[String], // filter columns
+      LogicalRelation,
+      HadoopFsRelation,
+      FileIndex)
+
+  def unapply(plan: LogicalPlan): Option[returnType] = plan match {
+    case project @ Project(
+          _,
+          filter @ Filter(
+            condition: Expression,
+            logicalRelation @ LogicalRelation(
+              fsRelation @ HadoopFsRelation(location, _, _, _, _, _),
+              _,
+              _,
+              _))) =>
+      val projectColumnNames = CleanupAliases(project)
+        .asInstanceOf[Project]
+        .projectList
+        .map(_.references.map(_.asInstanceOf[AttributeReference].name))
+        .flatMap(_.toSeq)
+      val filterColumnNames = condition.references.map(_.name).toSeq
+
+      Some(
+        project,
+        filter,
+        projectColumnNames,
+        filterColumnNames,
+        logicalRelation,
+        fsRelation,
+        location)
+
+    case filter @ Filter(
+          condition: Expression,
+          logicalRelation @ LogicalRelation(
+            fsRelation @ HadoopFsRelation(location, _, _, _, _, _),
```

Comment:
looks like `location: FileIndex` is not being used in this file. Can we remove it?

Reply:
Thanks for catching it.  Removed it.